packages/eth-network/src/networkManager.js

```diff
@@ -97,68 +102,71 @@ class NetworkManager {
     this.onSdkDisposedCallback = callback
   }
 
-  async reconnectNetwork() {
-    this.setNetwork(this.network)
+  hasDuplicatedNetwork(chainId, rpcUrl) {
+    return !this.networks.every(net => net.chainId !== chainId && net.url !== rpcUrl)
   }
 
-  async setNetwork(network, { force, redirect = true, notify = true } = {}) {
-
-    redux.dispatch('ACTIVE_CUSTOM_NETWORK', network)
+  async requestMetaMaskRPC(networkInfo) {
+    const metaMaskClient = this.browserExtension.ethereum
+    const currentChainId = this.browserExtension.getChainId
+      ? await this.browserExtension.getChainId()
+      : await metaMaskClient.request({ method: 'eth_chainId' })
+    if (currentChainId === networkInfo.chainId) return
+    if (!networkInfo.chainId) {
+      notification.error(t('network.custom.err'), t('network.custom.lackChainId'))
+      return 
+    }
+    const hexChainId = `0x${(+networkInfo.chainId).toString(16)}`
```

Comment:
can use the util method of ethers
https://docs.ethers.io/v5/api/utils/bytes/#utils-hexlify